### PR TITLE
Version3_Iteration_Plan_Week3

### PR DESCRIPTION
**Pull request by IonutCicio**
I have a real problem dealing with the keyboard, beacouse there are too many objects of the class "key", and the program starts lagging. How can I deal with this issue?

* I'm sorry if some variable identificators aren't the "best" ones, but, those, for the moment, are just some tests, I'm gonna change them later.

* I'm sorry for not putting any comment too, I'm still working on it.

### CODE REVIEW

**Comment by wbigger** [comment]
About identifiers (not "identificators"), don't worry, do your best, refactoring is an endless process, don't pursue perfection but follow the "just good enough" principle.

Also for comments, start commenting the lines that you use often, that contains some core behavior or particularly complex stuff (for example nested if). Redundant or obvious comments are just a waste of time for both code writer and reader.

About lagging issue... welcome to the wonderful world of optimization, where the pureness of computer science crashes against dirty things as device fragmentation, CPU and RAM performances, and user experience (UX). I have to analyze the code for more detailed hints, but as a rule of thumbs, push all the possible operations to initialization. For example, I see a "loadFont" into the Box show function, that is awful: reading from hard disk is one of the heaviest operations, and you do it hundreds of time per seconds 😨

**Comment by IonutCicio** [comment]
Thaks!! I just put the "loadfont" and the "textAllign" in the constructor of "Box" and now everything works perfectly! thanks again for the suggestion!

**Comment by IonutCicio** [comment]
Now I just have to Put somewhere the documentation about the wireframe layout of the "settings menu" and the "keyboard" and I can merge this branch to the master branch.

**Comment by wbigger** [comment]
I answered your last question in issue #9.

File: KeyboardTeacher/Box.pde [hunk 1]
```diff
@@ -12,6 +12,8 @@ class Box {
     selfHeight = H;
     text = T;
     textSize = selfWidth / text.length();
+    textAlign(CENTER, CENTER);
```
Comment (wbigger) [inline KeyboardTeacher/Box.pde:15]: Be aware that this setup instruction may be overwritten before the relative drawing function. That is while the local variables (in this case, `textFont`) live inside the objects, and no one can change them, the `textAlign()` function modifies global variables that anyone else can change. Is it clear?

In other words, you should put all the formatting functions just before the relative rendering functions. In this case, the rendering function is `text()`, so you should restore the textAlign function shortly before it.

**Pull request merged**